### PR TITLE
Adapt SyncBN API from Other's Work

### PR DESCRIPTION
**Pull request by zhanghang1989**
Adapt the excellent API from https://github.com/vacancy/Synchronized-BatchNorm-PyTorch

### CODE REVIEW

**Comment by acgtyrant** [comment]
I noticed that you replace `SharedTensor` API by this excellent API, so the latter is faster than the former?
